WaterLog_Backend/Models/DataPoints.cs

```diff
@@ -52,6 +52,17 @@ public List<Y> getv()
             }
             return outV;
         }
+
+        public List<T> getv2()
+        {
+            List<T> outV = new List<T>();
+            foreach (DataNode<T, Y> listV in dataPoints)
+            {
+                outV.Add(listV.x);
+
```

Comment:
There are many whitespaces in the code, please remove them.

Reply:
done all